packages/core/src/Core.ts

```diff
@@ -0,0 +1,53 @@
+import find from "lodash.find";
+import console from "loglevel";
+import { AeroGearConfiguration, ServiceConfiguration } from "./config";
+import { MetricsService } from "./metrics";
+/**
+ * AeroGear services core class.
+ * Defines internal api and helpers to be used by all top level SDK's
+ */
+export class AgsCore {
+
+  public configurations?: ServiceConfiguration[];
+  private metrics?: MetricsService;
+  /**
+   * Initialize all AeroGear services SDK's
+   *
+   * @param config configuration that should be injected to all available SDK's
+   */
+  public init(config: AeroGearConfiguration): Promise<void> {
+    if (!config || !config.services || config.services.length === 0) {
+      return Promise.reject("Invalid configuration format");
+    }
+    this.configurations = config.services;
+
+    return new Promise((resolve, reject) => {
+      this.metrics = new MetricsService();
```

Comment:
I know this is how it's done in other SDKs, but I am wondering given this is javascript and we get event emitters for free, should we not considering let the core emit a "ready" event, and then use the app module to init the metrics module.

It also means that if there are other services want to be inited automatically in the future, we don't have to do it in the core.

Reply:
Yes. We will also need to do the same for Push SDK. SDK init is just initialization of the configuration so events may be little bit too much, however I will experiment with this when initializing Push SDK. Push SDK work will allow us to even more improve this and use events.

Reply:
If we settle on the singleton pattern it might be even better to have the init call idempotent and have the promise resolve immediately when core is ready. This way we avoid corner cases of late subscribers to the event emitter.

Adding small code example in follow up comment